isic/ingest/services/publish/__init__.py

```diff
@@ -94,7 +101,62 @@ def cohort_publish(*, publish_request: PublishRequest) -> None:
     sync_elasticsearch_indices_task.delay_on_commit()
 
 
+def embed_iptc_metadata(blob: FieldFile, attribution: str, license_: str) -> Path:
+    # pyexiv2 operates on filenames directly, so we need to write to a temp file
+    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+        shutil.copyfileobj(blob.file, temp_file)
+        temp_file.flush()
+
+        with pyexiv2.Image(temp_file.name) as tmp_image:
+            tmp_image.modify_iptc(
+                {
+                    "Iptc.Application2.Copyright": "Copyright Creative Commons",
+                    "Iptc.Application2.Credit": attribution,
+                    "Iptc.Application2.Source": "ISIC Archive",
+                }
+            )
+            tmp_image.modify_xmp(
+                {
+                    "Xmp.xmpRights.WebStatement": LICENSE_URIS[license_],
+                    # necessary to create the "struct" for the licensor URL
+                    "Xmp.plus.Licensor": [""],
+                    "Xmp.plus.Licensor[1]/plus:LicensorURL": "https://www.isic-archive.com",
+                }
+            )
+    return Path(temp_file.name)
+
+
 def unembargo_images(*, qs: QuerySet[Image]) -> None:
-    for image in qs.iterator():
-        image.public = True
-        image.save(update_fields=["public"])
+    with transaction.atomic():
+        for image in qs.select_related("accession").iterator():
+            attribution = image.accession.attribution
+            license_ = image.accession.copyright_license
+
+            with (
+                embed_iptc_metadata(
+                    image.accession.blob,  # nosem: use-image-blob-where-possible
+                    attribution,
+                    license_,
+                ).open("rb") as sponsored_blob,
+                embed_iptc_metadata(image.accession.thumbnail_256, attribution, license_).open(
+                    "rb"
+                ) as sponsored_thumbnail_256_blob,
+            ):
+                image.accession.sponsored_blob = File(  # nosem: use-image-blob-where-possible
+                    sponsored_blob, name=f"{image.isic_id}.{image.extension}"
+                )
+                image.accession.sponsored_thumbnail_256_blob = File(
+                    sponsored_thumbnail_256_blob, name=f"{image.isic_id}_thumbnail.jpg"
+                )
+                image.accession.blob = ""  # nosem: use-image-blob-where-possible
```

Comment:
This should be garbage collected from the default storage, once the database commits.